### PR TITLE
ci: Add travis basic testing

### PR DESCRIPTION
**Pull request by jcvenegas**
Add a few tests to be run by travis.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: .ci/setup.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+source /etc/os-release
+if [ "$ID" == fedora ];then
+	sudo -E dnf -y install automake  bats
+elif [ "$ID" == ubuntu ];then
+	sudo add-apt-repository -y ppa:duggan/bats
```
Comment (jodh-intel) [inline .ci/setup.sh]: Why is this needed? `bats` is in the Ubuntu repos so is it a minimum version dependency issue?

File: image-builder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+From fedora:27
```
Comment (jodh-intel) [inline image-builder/Dockerfile]: Can this value come from a top-level `./versions.txt` to make it simpler to detect when this file needs to be updated?

File: image-builder/image_builder.sh [hunk 3]
```diff
@@ -64,14 +70,47 @@ shift $(( $OPTIND - 1 ))
...
 [ -n "${ROOTFS}" ] || usage
 [ -d "${ROOTFS}" ] || die "${ROOTFS} is not a directory"
+
+ROOTFS=$(readlink -f ${ROOTFS})
+IMAGE_DIR=$(dirname ${IMAGE})
+IMAGE_DIR=$(readlink -f ${IMAGE_DIR})
+IMAGE_NAME=$(basename ${IMAGE})
+
+
+
```
Comment (jodh-intel) [inline image-builder/image_builder.sh]: Nit: one blank line should be enough here.

File: .ci/run.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2017 Intel Corporation
```
Comment (jodh-intel) [inline .ci/run.sh]: 2018? (applies to all files)

File: .ci/run.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#!/bin/bash
```
Comment (jodh-intel) [inline .ci/run.sh]: This needs to be the very first line of the file to be honored.

File: .ci/setup.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (jodh-intel) [inline .ci/setup.sh]: Could you add a standard comment copyright header?

File: .ci/setup.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+source /etc/os-release
```
Comment (jodh-intel) [inline .ci/setup.sh]: Nit: worth pointing out that this won't work on Clear Linux.

File: tests/image_creation.bats [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	sudo ${image_builder_sh} -s ${image_size} -o "${tmp_dir}/image.img" "${rootfs}"
+}
+
+@test "Can create fedora image" {
+	build_image fedora
+}
+
+@test "Can create clearlinux image" {
+	build_image clearlinux
+}
```
Comment (jodh-intel) [inline tests/image_creation.bats]: Nit: missing blank line between these function.s

File: tests/image_creation.bats [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+{
+	distro="$1"
+	[ -n "$distro" ]
+	local rootfs="${tmp_dir}/rootfs-osbuilder"
+	sudo -E ${rootfs_sh} -r "${rootfs}" fedora
+	sudo ${image_builder_sh} -s ${image_size} -o "${tmp_dir}/image.img" "${rootfs}"
+}
+
+@test "Can create fedora image" {
+	build_image fedora
```
Comment (jodh-intel) [inline tests/image_creation.bats]: Neat! ;)

File: image-builder/Dockerfile (hunk 2 above)
Comment (jcvenegas) [reply image-builder/Dockerfile]: I am sorry but, Dockeriles does not support source from another file, (I can point to latests ?  and pray will work always)

Comment (jodh-intel) [reply image-builder/Dockerfile]: I guess you could use `latest`, or you could create `Dockerfile.in` so that it could then specify:

```
From: fedora:@fedora_docker_version@
```
or even,

```
From: @docker_distro_name@:@docker_distro_version@
```

... and do `sed` magic to translate variables from `versions.txt` when we "build" the `Dockerfile`.

File: .ci/setup.sh [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#Note: If add clearlinux as supported CI use a stateless os-release file
+source /etc/os-release
+
+if [ "$ID" == fedora ];then
+	sudo -E dnf -y install automake  bats
+elif [ "$ID" == ubuntu ];then
+	sudo add-apt-repository -y ppa:duggan/bats
```
Comment (jcvenegas) [inline .ci/setup.sh]: @jodh-intel  I need to add this repo because ubuntu from travis does not provide bats

File: image-builder/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,3 @@
+From fedora:latest
```
Comment (jcvenegas) [inline image-builder/Dockerfile]: @jodh-intel I changed to `latest`, I like Dockerfile templates but will take more time to re-work.

File: .ci/setup.sh (hunk 10 above)
Comment (jodh-intel) [reply .ci/setup.sh]: Ack. Could you add a comment explaining that bats isn't available for Ubuntu trusty then? That would make it easier to understand and if Travis ever offer Xenial, we then know we can drop this repo :)

File: image-builder/Dockerfile (hunk 11 above)
Comment (jodh-intel) [reply image-builder/Dockerfile]: OK, I guess that's fine for now. Once this lands, could you raise an issue so we don't forget about auto-generating the dockerfile though?

**Comment by jodh-intel** [comment]
lgtm

File: image-builder/Dockerfile (hunk 11 above)
Comment (jcvenegas) [reply image-builder/Dockerfile]: sure

**Comment by jcvenegas** [comment]
@bergwolf  @devimc @jbryce, @laijs @sameo  @teawater @weizhang555 we need one more approve to merge this, please take a look

**Comment by devimc** [comment]
lgtm
